integration_tests/test_upgrade.py

```diff
@@ -230,10 +228,9 @@ def exec(c, tmp_path_factory, testnet=True):
     c.supervisorctl("stop", "all")
 
 
-def test_cosmovisor_upgrade(testnet: Cronos, mainnet: Cronos, tmp_path_factory):
-    providers = [testnet, mainnet]
-    path = tmp_path_factory
-    with ThreadPoolExecutor(len(providers)) as executor:
-        as_completed(
```

Comment:
my bad, can we try
```
futures = [executor.submit(exec, net, path, net == testnet) for net in providers]
for future in as_completed(futures):
    result = future.result()
    if isinstance(result, Exception):
        pytest.fail(str(result))
```

Reply:
I don't know, i think it's maybe unnecessary, it also makes debugging more complex, and test case selection don't work with it.